csrc/ops/alias.cpp

```diff
@@ -840,4 +840,29 @@ TensorView* slice(
   return slice(inp, slices);
 }
 
+std::vector<TensorView*> split(
+    TensorView* in,
+    int64_t dim,
+    const int64_t num_slices) {
+  const auto in_logical = TensorDomain::noReductions(in->getLogicalDomain());
+  const auto num_dims = static_cast<int64_t>(in_logical.size());
+  dim = wrapDim(dim, num_dims);
+  Val* dim_size = in_logical[dim]->extent();
+  Val* slice_size = SimplifyingIrBuilder::divExpr(
+      dim_size, IrBuilder::create<Val>(num_slices));
```

Comment:
Should this be a `ceilDiv` expression instead? Consider splitting an axis of size 11 into 6 chunks. We would have 5 chunks of size 2 and one of size 1 using `ceilDiv`. Using `divExpr` we would have 5  chunks of size 1 and one of size 6. 

Using `ceilDiv` I think it's not always possible to return the requested number of chunks. The example here is illustrative: https://pytorch.org/docs/stable/generated/torch.chunk.html. In the last example there they are splitting an axis of size 13 into "6" chunks. If we use `divExpr` instead of `ceilDiv` here, then we'd use 5 chunks of size 2 then one of size _3_. However, using ceilDiv instead we'd have 4 chunks of size 3 then one of size 1.

I think it's probably fine to use `div` but we should clearly state the behavior for non-divisible chunking in the docs.

Reply:
I wasn't aware of `torch.chunk` -- thank you! 

Yes, that's indeed closest to what I've implemented. I intended to support divisible chunking only. But when the size is symbolic it's impossible to assert evenness or to avoid chunks of negative size. So I used `divExpr` here and documented the behavior in `alias.h`. 

If I use `ceilDiv` here, will nvFuser silently fail on negative-size slices?

Reply:
If negative-size slices are requested, you will be given zero-size slices i.e. empty tensors. I think going with `divExpr` makes sense let's just be clear to mention that the last tensor might have _more_ elements than all the others.

Reply:
FWIW, https://github.com/NVIDIA/Fuser/blob/1906d4f6eeca870e029e5ce8a646699b75043d68/csrc/dynamic_transform.cpp#L409-L414 errors out a negative size, which is good -- better than a silent error.

Reply:
Hmmmm. I thought we handled this by massaging the resize expressions themselves.  Essentially any integers are valid inputs to slice, and they get clipped appropriately, so I wonder why we are hitting a negative extent at this check...